containers/EmpSponsors.ts

```diff
@@ -18,56 +22,56 @@ interface EmpMap {
   [address: string]: SponsorMap;
 }
 
+interface FinancialContractQuery {
+  id: string;
+  sponsorPositions: PositionState;
+}
+
 const useEmpSponsors = () => {
   const { block$ } = Connection.useContainer();
   const { contract: emp } = EmpContract.useContainer();
+  const { loading, error, data } = useQuery(ACTIVE_POSITIONS);
 
   const [activePositions, setActivePositions] = useState<EmpMap>({});
 
   // get position information about every sponsor that has ever created a position.
   const querySponsors = async () => {
     if (emp) {
-      const newSponsorFilter = emp.filters.NewSponsor(null);
-      const newSponsorEvents = await emp.queryFilter(newSponsorFilter);
-
-      // Create a clone of the active-positions mapping that we can update.
-      let newPositions = { ...activePositions };
-
-      // Map all active sponsor information to each EMP.
-      if (!newPositions[emp.address]) {
-        newPositions[emp.address] = {};
+      if (error) {
+        console.error(`Apollo client failed to fetch graph data:`, error);
       }
+      if (!loading && data) {
+        // Create a clone of the active-positions mapping that we can update.
+        let newPositions = { ...activePositions };
```

Comment:
I may be misunderstanding something, but why do we have to start with a clone of the old state? What if a sponsor's position goes from nonzero to 0, will that sponsor continue showing up with their outdated values because they no longer show up in the graph query?

Reply:
I'm actually curious about this too. Why not start with a fresh state?

Reply:
Actually I'll try this out because I noticed that when switching the EMP's from the dropdown, the table would temporarily show new data and then disappear.

Reply:
Made these changes:
- Removed EMPMap, activePositions is now just a SponsorMap since we are only dealing with one EMP at a time.
- Each call to queryPositions starts with a fresh newPositions object.